stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt

```diff
@@ -182,64 +351,154 @@ private fun OnLastVisibleItemChanged(lazyListState: LazyListState, onChanged: (f
  */
 @Composable
 internal fun BoxScope.DefaultMessagesHelperContent(
     messagesState: MessagesState,
     lazyListState: LazyListState,
+    scrollToBottom: (() -> Unit) -> Unit,
 ) {
-    val (_, _, _, messages, _, _, newMessageState) = messagesState
+    val messages = messagesState.messageItems
+    val newMessageState = messagesState.newMessageState
+    val areNewestMessagesLoaded = messagesState.startOfMessages
+    val isMessageInThread = messagesState.parentMessageId != null
+
     val coroutineScope = rememberCoroutineScope()
 
     val firstVisibleItemIndex = lazyListState.firstVisibleItemIndex
 
     val focusedItemIndex = messages.indexOfFirst { it is MessageItemState && it.focusState is MessageFocused }
 
-    val offset = messagesState.focusedMessageOffset.collectAsState()
+    val offset by messagesState.focusedMessageOffset.collectAsState()
 
-    LaunchedEffect(
-        newMessageState,
-        firstVisibleItemIndex,
-        focusedItemIndex,
-        offset.value
-    ) {
-        if (focusedItemIndex != -1 && !lazyListState.isScrollInProgress) {
+    LaunchedEffect(newMessageState, focusedItemIndex, offset) {
+
+        if (focusedItemIndex != -1 &&
+            !lazyListState.isScrollInProgress
+        ) {
             coroutineScope.launch {
-                lazyListState.scrollToItem(focusedItemIndex, offset.value ?: 0)
+                lazyListState.scrollToItem(focusedItemIndex, offset ?: 0)
             }
         }
 
-        when {
-            !lazyListState.isScrollInProgress && newMessageState == Other &&
-                firstVisibleItemIndex < 3 -> coroutineScope.launch {
-                lazyListState.animateScrollToItem(0)
-            }
+        val shouldScrollToBottomOnNewMessage = shouldScrollToBottomOnNewMessage(
+            focusedItemIndex,
+            firstVisibleItemIndex,
+            newMessageState,
+            areNewestMessagesLoaded,
+            lazyListState.isScrollInProgress
+        )
 
-            !lazyListState.isScrollInProgress && newMessageState == MyOwn -> coroutineScope.launch {
-                if (firstVisibleItemIndex > 5) {
+        if (shouldScrollToBottomOnNewMessage) {
+            coroutineScope.launch {
+                if (newMessageState == MyOwn && firstVisibleItemIndex > 5) {
                     lazyListState.scrollToItem(5)
                 }
                 lazyListState.animateScrollToItem(0)
             }
         }
     }
 
-    if (abs(firstVisibleItemIndex) >= 3) {
+    if (isScrollToBottomButtonVisible(isMessageInThread, firstVisibleItemIndex, areNewestMessagesLoaded)) {
         MessagesScrollingOption(
             unreadCount = messagesState.unreadCount,
             modifier = Modifier.align(Alignment.BottomEnd),
             onClick = {
-                coroutineScope.launch {
-                    if (firstVisibleItemIndex > 5) {
-                        lazyListState.scrollToItem(5)
+                scrollToBottom {
```

Comment:
This is also a UX breaking change - we had the implementation of scrollign to the 5th item from the bottom then all the way to the end, as a nice easing scroll, rather than just jumping to the bottom. Can we keep that?

Reply:
we can keep it, will revert.